plugins/api/src/atomic_apply_metadata_operations.cpp

```diff
@@ -271,17 +305,25 @@ namespace
         return false;
     }
 
-    auto insert_metadata(nanodbc::connection& _db_conn, const fs::metadata& _metadata) -> int
+    auto insert_metadata(nanodbc::connection& _db_conn, const json& _db_plugin_config, const fs::metadata& _metadata) -> int
     {
         nanodbc::statement stmt{_db_conn};
 
-#ifdef ORA_ICAT
-        prepare(stmt, "insert into R_META_MAIN (meta_id, meta_attr_name, meta_attr_value, meta_attr_unit, create_ts, modify_ts) "
-                      "values (select R_OBJECTID.nextval from DUAL, ?, ?, ?, ?, ?)");
-#else
-        prepare(stmt, "insert into R_META_MAIN (meta_id, meta_attr_name, meta_attr_value, meta_attr_unit, create_ts, modify_ts) "
-                      "values (nextval('R_OBJECTID'), ?, ?, ?, ?, ?)");
-#endif // ORA_ICAT
+        if (_db_plugin_config.contains("oracle")) {
```

Comment:
at the very least, we are paying the prices of a lookup and string comparison multiple times.